lib/lotus/commands/generate.rb

```diff
@@ -1,93 +1,108 @@
         end
       end
 
-      # @since 0.3.0
-      # @api private
-      def spec_root
-        @spec_root ||= Pathname.new('spec')
-      end
 
-      # @since 0.3.1
-      # @api private
-      def model_root
-        @model_root ||= Pathname.new(['lib', ::File.basename(Dir.getwd)]
-          .join(::File::SEPARATOR))
+      desc 'generate migration NAME', 'generate a migration'
+      long_desc <<-EOS
+      `lotus generate migration` will generate an empty migration file.
+
+      > $ lotus generate migration do_something
+      EOS
+      def migration(name)
+        if options[:help]
+          invoke :help, ['migration']
+        else
+          require 'lotus/commands/generate/migration'
+          Lotus::Commands::Generate::Migration.new(options, name).start
+        end
       end
 
-      private
-      # @since 0.3.0
-      # @api private
-      def generator
-        require "lotus/generators/#{ @type }"
-        class_name = Utils::String.new(@type).classify
-        Utils::Class.load!(GENERATORS_NAMESPACE % class_name).new(self)
+      desc 'generate model NAME', 'generate an entity'
+      long_desc <<-EOS
+        `lotus generate model` will generate an entity along with repository
+        and corresponding tests. The name of the model can contain slashes to
+        indicate module names.
+
+        > $ lotus generate model car
+
+        > $ lotus generate model vehicles/car
+      EOS
+      method_option :test, desc: 'Defines the testing Framework to be used. Default is defined through your .lotusrc file.'
+      def model(name)
+        if options[:help]
+          invoke :help, ['model']
+        else
+          require 'lotus/commands/generate/model'
+          Lotus::Commands::Generate::Model.new(options, name).start
+        end
       end
 
-      def sanitize_input(app_name, name)
-        if options[:architecture] == APP_ARCHITECTURE
-          @app_name = nil
-          @name     = app_name
+      desc 'generate slice APPLICATION_NAME', 'generate a slice'
+      method_option :application_base_url, desc: 'Base URL for the new slice. If missing, then it is inferred from APPLICATION_NAME'
+      long_desc <<-EOS
+        `lotus generate slice` creates a new slice inside the 'apps' directory.
+
+        It can only be called for lotus applications with container architecture.
+
+        > $ lotus generate slice admin
+
+        > $ lotus generate slice reporting --url=/reports
+      EOS
+      def slice(application_name)
+        if options[:help]
+          invoke :help, ['slice']
         else
-          @app_name = app_name
-          @name     = name
+          require 'lotus/commands/generate/slice'
+          Lotus::Commands::Generate::Slice.new(options, application_name).start
         end
       end
+
+      # desc 'generate app APPLICATION', 'generate an app'
+      # method_option :url, desc: 'Missing', default: nil, type: :string, required: false
+      #
+      # def app(application_name)
```

Comment:
Yeah... since _app_ and _slice_ generator are basically the same i wanted to wait for your feedback on _app_ vs. _slice_ naming.

It looks like i'll remove this, rename the _slice_ generator to _app_ and be done with it.